syntax = "proto3";
package bcos.protocol;

// Mainly used for block data packet encoding when network transmission
message PBRawBlock{
    int32 version = 1;
    bytes header = 2;
    repeated bytes transactions = 3;
    repeated bytes receipts = 4;
    // In some scenarios, only transaction hashes are included in the block
    repeated bytes transactionsHash = 5;
    repeated bytes receiptsHash = 6;
    // the block type, include: 
    // CompleteBlock, OnlyWithTransactionsHash
    int32 type = 7;
}
